packages/abi-utils/lib/signature.ts

```diff
@@ -0,0 +1,46 @@
+import { Parameter, FunctionEntry, EventEntry, ErrorEntry } from "./types";
+import { soliditySha3 } from "web3-utils";
+
+export const ShortSelectorSize = 4;
+
+//NOTE: this function returns the written out SIGNATURE, not the SELECTOR
+export function abiSignature(
+  abiEntry: FunctionEntry | EventEntry | ErrorEntry
+): string {
+  return abiEntry.name + abiTupleSignature(abiEntry.inputs);
+}
+
+export function abiTupleSignature(parameters: Parameter[]): string {
+  const components = parameters.map(abiTypeSignature);
+  return "(" + components.join(",") + ")";
+}
+
+export function abiTypeSignature(parameter: Parameter): string {
+  const tupleMatch = parameter.type.match(/tuple(.*)/);
+  if (tupleMatch === null) {
+    //does not start with "tuple"
```

Comment:
The comment says does not start with `tuple` but the regex doesn't use `^`. Is that correct?

Reply:
Oops!  Good catch.  This code will work anyway, because there's no other place that `tuple` can occur, but I should change it to not rely on that, thanks.